dirty_cat/datasets/fetching.py

```diff
@@ -0,0 +1,294 @@
+
+        ),
+    )
+)
+
+EMPLOYEE_SALARIES_CONFIG = DatasetInfo(
+    name='employee_salaries',
+    urlinfos=(
+        UrlInfo(
+            url="https://data.montgomerycountymd.gov/api/views/"
+                "xj3h-s2i7/rows.csv?accessType=DOWNLOAD",
+            filenames=("rows.csv",),
+            uncompress=False
+        ),
+    )
+)
+
+TRAFFIC_VIOLATIONS_CONFIG = DatasetInfo(
+    name='traffic_violations',
+    urlinfos=(
+        UrlInfo(
+            url="https://data.montgomerycountymd.gov/api/views/"
+                "4mse-ku6q/rows.csv?accessType=DOWNLOAD",
+            filenames=(
+                "rows.csv",
+            ), uncompress=True
+        ),
+    )
+)
+
+FOLDER_PATH = os.path.dirname(os.path.realpath(__file__))
+
+
+class FileChangedError(Exception):
+    pass
+
+
+def _download_and_write(url, file, show_progress=True):
+    if show_progress:
+        from clint.textui import progress
+    try:
+        # using stream=True to download the response body only when
+        # accessing the content attribute
+        with request_get(url, stream=True) as r:
+            total_length = r.headers.get('Content-Length')
+            if total_length is not None:
+                with open(file, 'wb') as local_file:
+                    content_iterator = r.iter_content(chunk_size=1024)
+                    if show_progress:
+                        content_iterator = progress.bar(
+                            content_iterator, expected_size=(int(total_length) /
+                                                             1024) + 1)
+
+                    for chunk in content_iterator:
+                        if chunk:
+                            local_file.write(chunk)
+                            local_file.flush()
+
+            else:
+                warnings.warn('content size cannot be found, '
+                              'downloading file from {} as a whole'.format(
+                    url))
+                with open(file, 'wb') as local_file:
+                    local_file.write(r.content)
+
+    except requests.RequestException as e:
+        # pretty general request exception. subject to change
+        raise Exception('error while fetching: {}'.format(e))
+
+
+def fetch_dataset(configfile: DatasetInfo, show_progress=True):
+    data_dir = os.path.join(get_data_dir(), configfile.name)
+    for urlinfo in configfile.urlinfos:
+        _fetch_file(urlinfo.url, data_dir, filenames=urlinfo.filenames,
+                    uncompress=urlinfo.uncompress, show_progress=show_progress)
+    return data_dir
+
+
+def _fetch_file(url, data_dir, filenames=None, overwrite=False,
+                md5sum=None, uncompress=True, show_progress=True):
+    """fetches the content of a requested url
+
+    IF the downloaded file is compressed, then the fetcher
+    looks also for the uncompressed files before downloading .
+
+
+    Parameters
+    ----------
+    url
+    data_dir: directory where the data will be stored
```

Comment:
This is not exactly the numpy doc standard. Please have a look at the standard and follow it. This is important for rendering of the docs